prefab_cloud_python/logging.py

```diff
@@ -1,55 +1,88 @@
 import logging
-from typing import Optional
+from typing import Optional, Iterator, Any
 
 from structlog import DropEvent
 
 import prefab_cloud_python
 from prefab_cloud_python import Client
 
 
-def iterate_dotted_string(s: str):
+def iterate_dotted_string(s: str) -> Iterator[str]:
     parts = s.split(".")
     for i in range(len(parts), 0, -1):
         yield ".".join(parts[:i])
 
 
-class LoggerFilter(logging.Filter):
-    def __init__(self, client=None) -> None:
-        """Filter for use with standard logging. Will get its client reference from prefab_python_client.get_client() unless overridden"""
-        super().__init__()
+class BaseLoggerFilterProcessor:
+    def __init__(self, client: Client = None) -> None:
         self.client = client
 
-    def _get_client(self) -> "prefab_cloud_python.Client":
+    def _get_client(self) -> Client:
         if self.client:
             return self.client
         return prefab_cloud_python.get_client()
 
+    def _should_log_message(
+        self, client: Client, logger_name: str, called_method_level: int
+    ) -> bool:
+        closest_log_level = client.get_loglevel(logger_name)
+        return called_method_level >= closest_log_level
+
+
+class LoggerFilter(BaseLoggerFilterProcessor):
+    """Filter for use with standard logging. Will get its client reference from prefab_python_client.get_client() unless overridden"""
+
+    def __init__(self, client: Client = None) -> None:
+        super().__init__(client)
+
+    def logger_name(self, record: logging.LogRecord) -> str:
+        """Override this as needed to derive a different logger name"""
+        return record.name
+
     def filter(self, record: logging.LogRecord) -> bool:
         """this method is used with the standard logger"""
         client = self._get_client()
         if client:
-            client.record_log(record.name, record.levelno)
-            return self._should_log_message(client, record.name, record.levelno)
+            logger_name = self.logger_name(record)
+            if logger_name:
+                client.record_log(logger_name, record.levelno)
+                return self._should_log_message(client, logger_name, record.levelno)
         return True
 
-    def processor(self, logger, method_name: str, event_dict):
+
+class LoggerProcessor(BaseLoggerFilterProcessor):
+    """this class is for use with structlogger"""
+
+    def __init__(self, client: Client = None) -> None:
```

Comment:
`Optional[Client]` (or my preferred: `Client | None`)

Reply:
OOC why do you prefer Client | None ?